docs/publishers/authorization-grant-tokens.rst

```diff
@@ -0,0 +1,145 @@
+assume that:
+
+- Your :term:`Client ID` is ``4a2fa3b4-c160-4436-82d3-148f602c9aa8``.
+- Your authority is ``customwidgets.com``.
+- The user has a username of ``samina.mian``.
+- The current time is ``2016-11-08T11:35:45Z``, which corresponds to a UTC Unix
+  timestamp of ``1478601345``.
+- The token should be valid for about 10 minutes, i.e. until ``1478601945``,
+  expressed as a UTC Unix timestamp.
+- The token should be valid for the annotation service running at
+  ``hypothes.is``.
+
+With these data, we can construct a token payload. It should look like the
+following::
+
+   {
+     "aud": "hypothes.is",
+     "iss": "4a2fa3b4-c160-4436-82d3-148f602c9aa8",
+     "sub": "acct:samina.mian@customwidgets.com",
+     "nbf": 1478601345,
+     "exp:" 1478601945
+   }
+
+You should sign this payload using the ``HS256`` JWT-signing algorithm, using
+the :term:`Client secret` as the key. The result will look something like this::
+
+   eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiI0QTJGQTNCNC1DMTYwLTQ0MzYtODJEMy0xNDhGNjAyQzlBQTgiLCJuYmYiOjE0Nzg2MDEzNDUsImF1ZCI6Imh5cG90aGVzLmlzIiwiZXhwIjoxNDc4NjAxOTQ1LCJzdWIiOiJhY2N0OnNhbWluYS5taWFuQGN1c3RvbXdpZGdldHMuY29tIn0.65-ZErbLu1q8LpT_K8FAOQO984hAyN1XFBe1rC3lgfk
+
+.. seealso::
+
+   `RFC7523 <https://tools.ietf.org/html/rfc7523>`_, "JSON Web Token (JWT)
+   Profile for OAuth 2.0 Client Authentication and Authorization Grants". Note
+   that we currently only support the ``HS256`` signing algorithm, and not the
+   public-key ``RS256`` signing algorithm mentioned in the RFC.
+
+Examples
+--------
+
+This section contains complete example code for generating a JWT in various
+common programming environments.
+
+Python
+``````
+
+We recommend using `PyJWT <https://pyjwt.readthedocs.io/en/latest/>`_::
+
+   import datetime
+   import jwt
+
+   # IMPORTANT: replace these values with those for your client account!
+   CLIENT_AUTHORITY = 'customwidgets.com'
+   CLIENT_ID        = '4a2fa3b4-c160-4436-82d3-148f602c9aa8'
+   CLIENT_SECRET    = '5SquUVG0Tpg57ywoxUbPPgjtK0OkX1ttipVlfBRRrpo'
+
+   def generate_grant_token(username):
+      now = datetime.datetime.utcnow()
+      userid = 'acct:{username}@{authority}'.format(username=username,
+                                                    authority=CLIENT_AUTHORITY)
+      payload = {
+         'aud': 'hypothes.is',
+         'iss': CLIENT_ID,
+         'sub': userid,
+         'nbf': now,
+         'exp': now + datetime.timedelta(minutes=10),
+      }
+      return jwt.encode(payload, CLIENT_SECRET, algorithm='HS256')
+
+Ruby
+````
+
+We recommend using `ruby-jwt <https://jwt.github.io/ruby-jwt/>`_::
+
+   require 'jwt'
+
+   # IMPORTANT: replace these values with those for your client account!
+   CLIENT_AUTHORITY = 'customwidgets.com'
+   CLIENT_ID        = '4a2fa3b4-c160-4436-82d3-148f602c9aa8'
+   CLIENT_SECRET    = '5SquUVG0Tpg57ywoxUbPPgjtK0OkX1ttipVlfBRRrpo'
+
+   def generate_grant_token(username)
+     now = Time.now.to_i
```

Comment:
I don't think it makes any difference. `to_i` turns the time into the number of seconds since the epoch, which is defined as UTC.